test/asn1.test.js

```diff
@@ -0,0 +1,117 @@
+// Testing lib
+import * as chai from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+
+import crypto from "crypto";
+import { Fido2Lib } from "../lib/main.js";
+
+chai.use(chaiAsPromised.default);
+const assert = chai.assert;
+
+const sha256 = (data) => {
+	return crypto.createHash("sha256").update(data).digest("hex");
+};
+
+const base64url = (buf) => {
+	return Buffer.from(buf).toString("base64url");
+};
+const base64urlToBuffer = (str) => {
+	return Buffer.from(str, "base64url");
+};
+const base64urlToArrayBuffer = (str) => {
+	const buf = base64urlToBuffer(str);
+	return buf.buffer.slice(buf.byteOffset, buf.byteOffset + buf.byteLength);
+};
+
+const generateSignature = (authenticatorData, clientData, privateKey) => {
+	const dataHash = sha256(clientData);
+	const authenticatorDataHex = base64urlToBuffer(authenticatorData).toString("hex");
+	let signature;
+	do {
+		signature = new Uint8Array(
+			crypto.sign(
+				null,
+				Buffer.from(authenticatorDataHex + dataHash, "hex"),
+				privateKey
+			)
+		);
+	} while(signature[3] >= 32);
+
+	return base64url(signature);
+};
+
+const generateAuthenticatorData = () => {
```

Comment:
Is it really necessary to generate this input dynamically? Usually tests run on real sample outputs. Please try to follow the other test templates and preferably put your test into that.

Reply:
Updated the tests to use static values.